x-pack/legacy/plugins/apm/server/routes/transaction.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as t from 'io-ts';
+import { setupRequest } from '../lib/helpers/setup_request';
+import { getRootTransactionByTraceId } from '../lib/transactions/get_transaction_by_trace';
+import { createRoute } from './create_route';
+
+export const transactionByTraceIdRoute = createRoute(() => ({
+  path: '/api/apm/transaction/{traceId}',
```

Comment:
@cauemarcondes It just occurs to me that this route is incorrect - at least if we were to follow restful principles.

I'd expect a url like "/api/apm/transaction/123" to return the transaction with the identifier "123". That's not the case here.

A better route would be `/api/apm/traces/{traceId}/root_transaction`. And in that case it should be moved to https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/apm/server/routes/traces.ts#L19.

Not important but feel free to change this if you have time at some point.

Reply:
@sqren I've just seen this message now. So I created a new ticket #52252.